src/test/scala/it/pagopa/interop/authorizationprocess/util/SpecUtils.scala

```diff
@@ -237,6 +271,18 @@ trait SpecUtils extends SprayJsonSupport { self: MockFactory =>
     updatedAt = None
   )
 
+  val archivedPurpose: PurposeManagementDependency.Purpose =
+    purpose.copy(
+      id = UUID.randomUUID(),
+      versions = Seq(purposeVersion.copy(state = PurposeManagementDependency.PurposeVersionState.ARCHIVED))
+    )
+
+  val notArchivedPurpose: PurposeManagementDependency.Purpose =
+    purpose.copy(
+      id = UUID.randomUUID(),
+      versions = Seq(purposeVersion.copy(state = PurposeManagementDependency.PurposeVersionState.ACTIVE))
```

Comment:
I suggest to add this case to improve testing
```suggestion
      versions = Seq(purposeVersion.copy(state = PurposeManagementDependency.PurposeVersionState.ACTIVE), purposeVersion.copy(state = PurposeManagementDependency.PurposeVersionState.ARCHIVED))
```

Reply:
But it must be not in Archived STATE, in this case.